upstream/upstream.go

```diff
@@ -655,10 +667,102 @@ func (u *Upstream) shouldSkip(req *common.NormalizedRequest) (reason error, skip
 	}
 
 	// TODO evm: if block can be determined from request and upstream is only full-node and block is historical skip
+	jrReq, err := req.JsonRpcRequest()
+	if err != nil {
+		return fmt.Errorf("failed to get json-rpc request: %w", err), false
+	}
+
+	// sample of methods with block param
+	methodsWithBlockParam := map[string]int{
+		"eth_getBalance":          1,
+		"eth_getCode":             1,
+		"eth_getTransactionCount": 1,
+		"eth_call":                1,
+	}
```

Comment:
It's better to move these to common/evm.go as well, we're using similar info in evm block extraction, so let's see what are the ways to have this list in once place...

Also let's define this array as global const so it's not created per request, it's unnecessary memory usage